core/src/main/java/io/micronaut/core/io/IOUtils.java

```diff
@@ -140,7 +140,11 @@ static Path resolvePath(@NonNull URI uri,
                 if (!jarUri.startsWith(SCHEME_FILE + COLON)) {
                     // Special case WebLogic classloader
                     // https://github.com/micronaut-projects/micronaut-core/issues/8636
-                    jarUri = SCHEME_FILE + COLON + jarUri;
+					if (jarUri.startsWith("/")) {
```

Comment:
If we want to narrow down this condition much more (to handle specifically the scheme-specific part starting with "C:" or similar, I can change this condition to check that jarUri[0] is alpha and jarUri[1] is `:`.